project/RamlTypeGenerator.scala

```diff
@@ -481,21 +484,26 @@ object RamlTypeGenerator {
         actualFields.map(_.toString).exists(t => t.toString.startsWith(name) || t.toString.contains(s"[$name]"))) {
         actualFields.map(_.constraints).requiredImports ++ Seq(
           OBJECTDEF("playJsonFormat") withParents PLAY_JSON_FORMAT(name) withFlags Flags.IMPLICIT := BLOCK(
-            DEF("reads", PLAY_JSON_RESULT(name)) withParams PARAM("json", PlayJsValue) := BLOCK(
-              actualFields.map { field =>
-                VAL(field.name) := field.playValidator
-              } ++ Seq(
-                VAL("_errors") := SEQ(actualFields.map(f => TUPLE(LIT(f.rawName), REF(f.name)))) DOT "collect" APPLY BLOCK(
-                  CASE(REF(s"(field, e:$PlayJsError)")) ==> (REF("e") DOT "repath" APPLY(REF(PlayPath) DOT "\\" APPLY REF("field"))) DOT s"asInstanceOf[$PlayJsError]"),
-                IF(REF("_errors") DOT "nonEmpty") THEN (
-                  REF("_errors") DOT "reduceOption" APPLYTYPE PlayJsError APPLY (REF("_") DOT "++" APPLY REF("_")) DOT "getOrElse" APPLY (REF("_errors") DOT "head")
-                  ) ELSE (
-                  REF(PlayJsSuccess) APPLY (REF(name) APPLY
-                    actualFields.map { field =>
-                      REF(field.name) := (REF(field.name) DOT "get")
-                    }))
+            DEF("reads", PLAY_JSON_RESULT(name)) withParams PARAM("json", PlayJsValue) := {
+              if (serializeOnly) BLOCK(
+                THROW( NEW( REF(s"NotImplementedError") APPLY LIT("This type can only be de-serialized") ) )
```

Comment:
Actually it would be nicer if no reads function would be generated in this case – instead of generating one that throws. Could be changed later on if we feel like it.

Reply:
It's a requirement from the `play.api.libs.json.Format` interface, where the `reads` and `writes` are exposed. It cannot be blank :disappointed:

Reply:
Yes and no. The generator is written to create a
```
implicit object playJsonFormat extends play.api.libs.json.Format[Whatever]
```

A `Format` requires both `Reads` and `Writes`, but the generator could as well only create a
```
implicit object playJsonReads extends play.api.libs.json.Reads[Whatever]
```

meaning no `Writes` and no `Format`. As I said,
> Could be changed later on if we feel like it.

I don't see a strong reason now to mess with the generator if we're eventually replacing it.

Reply:
True, I was considering performing the minimum possible interventions in the code since I consider every change there quite fragile and untested :smile: 

But I don't think changing `Format` to `Writes` would cause much harm..